src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp/PersistingAuthenticationStateProvider.cs

```diff
@@ -0,0 +1,51 @@
+using BlazorWeb_CSharp.Client;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Options;
+
+namespace BlazorWeb_CSharp;
+
+public class PersistingAuthenticationStateProvider : AuthenticationStateProvider, IDisposable
```

Comment:
Implement `IHostEnvironmentAuthenticationStateProvider` and Blazor will populate the initial state from HttpContext for you (and you don't need to use the accessor).